implementation/src/main/java/io/smallrye/health/AsyncHealthCheckFactory.java

```diff
@@ -51,12 +52,13 @@ private HealthCheckResponse handleFailure(String name, Throwable e) {
         HealthLogging.logger.healthCheckError(e);
 
         HealthCheckResponseBuilder response = HealthCheckResponse.named(name).down();
-
         switch (uncheckedExceptionDataStyle) {
             case ROOT_CAUSE:
+                response.withData(EXCEPTION, e.toString());
```

Comment:
I'm not completely sure whether `toString()` is actually a good idea, you never know how the user might have implemented that, and it could get quite crazy. Wouldn't it be better to have two separate fields, one with the exception class, one with the message? 
I mean, by default, `toString()` returns both of these things, but you never know...